scoring/src/main/scala/ai/h2o/sparkling/ml/models/H2OMOJOPredictionMultinomial.scala

```diff
@@ -0,0 +1,53 @@
+/*
+* Licensed to the Apache Software Foundation (ASF) under one or more
+* contributor license agreements.  See the NOTICE file distributed with
+* this work for additional information regarding copyright ownership.
+* The ASF licenses this file to You under the Apache License, Version 2.0
+* (the "License"); you may not use this file except in compliance with
+* the License.  You may obtain a copy of the License at
+*
+*    http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+package ai.h2o.sparkling.ml.models
+
+import ai.h2o.sparkling.ml.models.H2OMOJOPredictionMultinomial.Base
+import org.apache.spark.sql.expressions.UserDefinedFunction
+import org.apache.spark.sql.functions.{col, udf}
+import org.apache.spark.sql.types.{ArrayType, DoubleType, StructField}
+import org.apache.spark.sql.{Column, Row}
+
+trait H2OMOJOPredictionMultinomial {
+  self: H2OMOJOModel =>
+  def getMultinomialPredictionUDF(): UserDefinedFunction = {
+    udf[Base, Row] { r: Row =>
+      val pred = easyPredictModelWrapper.predictMultinomial(RowConverter.toH2ORowData(r))
+      Base(pred.classProbabilities)
+    }
+  }
+
+  def getMultinomialPredictionColSchema(): Seq[StructField] = {
+    getDetailedPredictionColSchema()
+  }
+
+  def getMultinomialDetailedPredictionColSchema(): Seq[StructField] = {
+    StructField("probabilities", ArrayType(DoubleType)) :: Nil
+  }
+
+
+  def extractMultinomialPredictionColContent(): Column = {
+    col(getDetailedPredictionCol())
```

Comment:
at the moment for most of the predictions the content of detailed prediction and just the prediction is the same, so the calls above are actually returning the same. But true is that we can be more explicit and at explicitly return elements from detailed prediction ( even though we extract all). 

The benefit of this can be that we won't accidentally introduce new field to regular prediction. Anyway though, the goal is to return just single value in the prediction col

Reply:
Maybe, we could start prioritizing some features just for a major release. Right now, it's difficult to me to consider if such changes are bug free or not. In such changes in an intermediate state that are backward compatible, I can just believe that the implementation is exactly as you intended.

Reply:
Fine with me